src/lib/viewers/doc/DocBaseViewer.js

```diff
@@ -1596,6 +1596,11 @@ class DocBaseViewer extends BaseViewer {
 
     handleAnnotationControlsClick({ mode }) {
         const nextMode = this.annotationControlsFSM.transition(AnnotationInput.CLICK, mode);
+
+        if (nextMode === AnnotationMode.REGION) {
```

Comment:
`annotationControlsFSM` currently subscribes to the transition changes but calls `updateDiscoverabilityResinTag` directly, perhaps we could introduce a `handler` that also checks for this condition and applies the class there.

Also it seems like we would want to update localStorage on the transition from `REGION` to `NONE`, or even to `REGION` but only trigger the logic to apply the class on any subsequent transition to `REGION`

Reply:
Discussed offline that since the transition changes only happen when the user clicks, we're okay with not needing the handler.

Reply:
I'm still debating whether this should be moved out to a fsm callback. This method is pretty mode agnostic so I'm wary of cluttering it with some region specific code. Let's do the callback because that'll be easier to strip out eventually too since it'll be it's own separate function. Something like:
```
this.annotationControlsFSM.subscribe(this.applyCursorFtux);
```